gobblin-runtime/src/main/java/org/apache/gobblin/runtime/messaging/data/DynamicWorkUnitSerde.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.runtime.messaging.data;
+
+import com.google.common.base.Charsets;
+import com.google.common.base.Preconditions;
+import com.google.gson.Gson;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import java.nio.charset.Charset;
+import lombok.NonNull;
+
+/**
+ * Library for serializing / deserializing {@link DynamicWorkUnitMessage}. This solution maintains implementation
+ * specific fields not specified in the interface and is dependent on {@link Gson} to do the serialization / deserialization
+ */
+public final class DynamicWorkUnitSerde {
+  private static final Gson GSON = new Gson();
+  private static final String PROPS_PREFIX = "DynamicWorkUnit.Props";
+  private static final String MESSAGE_IMPLEMENTATION = PROPS_PREFIX + ".MessageImplementationClass";
+  private static final Charset DEFAULT_CHAR_ENCODING = Charsets.UTF_8;
+
+  private DynamicWorkUnitSerde() {
+    throw new RuntimeException("Cannot instantiate utils class");
```

Comment:
This RuntimeException should be redundant if the constructor is private?

Reply:
Fair enough. I'll follow the pattern in http://hg.openjdk.java.net/jdk8u/jdk8u/jdk/file/be44bff34df4/src/share/classes/java/util/Arrays.java#l81 and https://github.com/apache/commons-collections/blob/master/src/main/java/org/apache/commons/collections4/CollectionUtils.java#L221-L224

This is addressed in the latest iteration